php/libraries/IdGenerator.class.inc

```diff
@@ -0,0 +1,1208 @@
+<?php
+/**
+ * Generates Ids used for PSCID and ExternalId
+ *
+ * PHP Version 5
+ *
+ * @category Main
+ * @package  Loris
+ * @author   anyhowstep <justin.ng.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+/**
+ * A collection of utility methods for id generation.
+ *
+ * PHP Version 5
+ *
+ * @category Main
+ * @package  Loris
+ * @author   anyhowstep <justin.ng.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+class IdGeneratorUtil
```

Comment:
This file has, like, 6 classes in it or something. I could split it up but they're really all serving one purpose. I'm okay with whatever you decide.